pid_controller/src/pid_controller.cpp

```diff
@@ -496,7 +496,15 @@ controller_interface::return_type PidController::update_and_write_commands(
   {
     for (size_t i = 0; i < measured_state_values_.size(); ++i)
     {
-      measured_state_values_[i] = state_interfaces_[i].get_value();
+      const auto state_interface_value_op = state_interfaces_[i].get_optional();
+      if (!state_interface_value_op.has_value())
+      {
+        RCLCPP_DEBUG(
+          get_node()->get_logger(), "Unable to retrieve the state interface value for %s",
+          state_interfaces_[i].get_name().c_str());
+        return controller_interface::return_type::OK;
```

Comment:
should we really return here? if not, it will just use the old measured_state_values_ right?

Reply:
Maybe with PID it is not a big issue? 
Usually, in my use case, the reference comes at lower frequency and the PID runs at high freq trying to follow the commands

Reply:
yes, if we are not returning it will use the old measured_state_values_ and PID update cycle will continue with previous value for that DOF. As @saikishor  mentioned PID runs at high freq, it can accommodate even with old value. So, I also think we don’t need to return early, as that might interrupt the PID update cycle. What do you think ?

```suggestion
        continue ;
```

Reply:
Yes that is what I was thinking about (run the update with the old value)

Reply:
Sounds good